File: Project Templates/VIPER Single View App.xctemplate/Objective-C/FirstPresenter.m
//___FILEHEADER___
 
#import "FirstPresenter.h"

@implementation FirstPresenter

- (instancetype)initWithInterface:(id<FirstViewProtocol>)interface interactor:(id<FirstInteractorInputProtocol>)interactor router:(id<FirstWireframeProtocol>)router
{
    if (self = [super init])
    {
        self.view = interface;
        self.interactor = interactor;
        self.router = router;
        [self.interactor setPresenter:self];
    }
    return self;
}

@end

File: Project Templates/VIPER Single View App.xctemplate/FirstPresenter.swift
//___FILEHEADER___

import UIKit

class FirstPresenter: FirstPresenterProtocol  {
    
    weak var view: FirstViewProtocol?
    var interactor: FirstInteractorInputProtocol?
    var wireFrame: FirstRouteProtocol?
    
    init() {}
    
}

extension FirstPresenter: FirstInteractorOutputProtocol {
    
    /// Extension INTERACTOR -> PRESENTER
    
}
